DQMOffline/Trigger/plugins/TopMonitor.cc

```diff
@@ -0,0 +1,973 @@
+  , workingpoint_(iConfig.getParameter<double>("workingpoint"))
+  //Suvankar
+  , usePVcuts_ ( iConfig.getParameter<bool>("applyleptonPVcuts")    )
+{
+
+    METME empty;
+    empty.numerator = nullptr;
+    empty.denominator = nullptr;
+
+    metME_ = empty ;   
+    metME_variableBinning_ = empty ;   
+    metVsLS_ = empty ;   
+    metPhiME_ = empty ;   
+    eventHT_ = empty ;   
+    jetVsLS_ = empty ; 
+    muVsLS_ = empty ; 
+    eleVsLS_ = empty ; 
+    // Marina
+    bjetVsLS_ = empty ;
+    htVsLS_ = empty ; 
+    jetEtaPhi_HEP17_ = empty; // for HEP17 monitoring
+
+    jetMulti_ = empty;
+    eleMulti_ = empty;
+    muMulti_ = empty;
+    // Marina
+    bjetMulti_ = empty;
+
+    elePt_jetPt_ = empty;
+    elePt_eventHT_ = empty;
+
+    ele1Pt_ele2Pt_ = empty;
+    ele1Eta_ele2Eta_ = empty;
+    mu1Pt_mu2Pt_ = empty;
+    mu1Eta_mu2Eta_ = empty;
+    elePt_muPt_ = empty;
+    eleEta_muEta_ = empty;
+
+    //BTV
+    DeltaR_jet_Mu_ = empty;
+
+    muPhi_= std::vector<METME> (nmuons_,empty);
+    muEta_= std::vector<METME> (nmuons_,empty);
+    muPt_= std::vector<METME> (nmuons_,empty);
+    muEta_variableBinning_= std::vector<METME> (nmuons_,empty);
+    muPt_variableBinning_= std::vector<METME> (nmuons_,empty);
+    muPtEta_= std::vector<METME> (nmuons_,empty);
+    muEtaPhi_= std::vector<METME> (nmuons_,empty);
+
+    elePhi_= std::vector<METME> (nelectrons_,empty);
+    eleEta_= std::vector<METME> (nelectrons_,empty);
+    elePt_= std::vector<METME> (nelectrons_,empty);
+    eleEta_variableBinning_= std::vector<METME> (nelectrons_,empty);
+    elePt_variableBinning_= std::vector<METME> (nelectrons_,empty);
+    elePtEta_= std::vector<METME> (nelectrons_,empty);
+    eleEtaPhi_= std::vector<METME> (nelectrons_,empty);
+
+
+    jetPhi_= std::vector<METME> (njets_,empty);
+    jetEta_= std::vector<METME> (njets_,empty);
+    jetPt_= std::vector<METME> (njets_,empty);
+    jetEta_variableBinning_= std::vector<METME> (njets_,empty);
+    jetPt_variableBinning_= std::vector<METME> (njets_,empty);
+    jetPtEta_= std::vector<METME> (njets_,empty);
+    jetEtaPhi_= std::vector<METME> (njets_,empty);
+
+    // Marina
+    bjetPhi_= std::vector<METME> (nbjets_,empty);
+    bjetEta_= std::vector<METME> (nbjets_,empty);
+    bjetPt_= std::vector<METME> (nbjets_,empty);
+    bjetCSV_= std::vector<METME> (nbjets_,empty);
+    bjetEta_variableBinning_= std::vector<METME> (nbjets_,empty);
+    bjetPt_variableBinning_= std::vector<METME> (nbjets_,empty);
+    bjetPtEta_= std::vector<METME> (nbjets_,empty);
+    bjetEtaPhi_= std::vector<METME> (nbjets_,empty);
+    bjetCSVHT_= std::vector<METME> (nbjets_,empty);
+
+  //Suvankar
+  lepPVcuts_.dxy = (iConfig.getParameter<edm::ParameterSet>("leptonPVcuts")).getParameter<double>("dxy");
+  lepPVcuts_.dz  = (iConfig.getParameter<edm::ParameterSet>("leptonPVcuts")).getParameter<double>("dz");     
+}
+
+TopMonitor::~TopMonitor()
+{
+    if (num_genTriggerEventFlag_) num_genTriggerEventFlag_.reset();
```

Comment:
You don't need to explicitly reset the unique_ptr. It will be done automatically.